fsw/cfe-core/CMakeLists.txt

```diff
@@ -37,6 +37,22 @@ endforeach(MODULE ${CFE_CORE_MODULES})
 
 add_library(${CFE_CORE_TARGET} STATIC ${CFE_ALL_MODULE_SRCS})
 
+foreach(MODULE ${MISSION_CORE_MODULES})
+  get_filename_component(MODULE_NAME ${MODULE} NAME)
+
+  # Search in core, then for custom module
+  if(IS_DIRECTORY ${cfe-core_MISSION_DIR}/src/${MODULE})
+    add_subdirectory(src/${MODULE} ${MODULE_NAME})
+  elseif(IS_DIRECTORY ${MISSION_SOURCE_DIR}/${MODULE})
+    add_subdirectory(${MISSION_SOURCE_DIR}/${MODULE} ${MODULE_NAME})
+  else()
+    message(FATAL_ERROR "Target build incomplete, ${MODULE} not found in ${MISSION_SOURCE_DIR} or ${cfe-core_MISSION_DIR}/src")
+  endif()
+
+  message (STATUS "Adding mission core module: " ${MODULE})
+  target_link_libraries(${CFE_CORE_TARGET} ${MODULE_NAME})
+endforeach(MODULE ${MISSION_CORE_MODULES})
+
```

Comment:
In my previous comments when I said "let the build system figure out the absolute path" ... I meant using the existing logic for doing so, not to add a new search function.

This message module shouldn't be treated specially or be any different than any other module, except for how its linked into the final product.

My previous commit in PR #720 covers this.  Can we look again/discuss integrating these and avoiding having another parallel module search implementation?

Reply:
I'm not a fan of: 

https://github.com/nasa/cFE/blob/d217ca39f1de74695f0d4e0eb427f90fbe8b1156/cmake/mission_build.cmake#L125-L171

I'd rather push off that discussion until we have an opportunity to more formally review and document.

Reply:
But that's all old/existing code -- it works for finding the apps and libs and everything else, even if they are out of tree.  Why wouldn't we just use it here?

PR #720 just reuses it for another module type, and I suggest doing the same here. 

If there is something wrong with it, I'm happy to fix/address that separately but we shouldn't add another implementation of the same thing.

Reply:
BTW, in addition to finding modules this is also responsible for setting the `${<modulename>_MISSION_DIR}` variable, which is important for any other module that depends on it so it can do e.g. `include_directories()` based on that.

The existing pattern is also to put things these user-specified modules in either `apps` or `libs` top level directories.  This appears to be allowing modules to exist as their own TLD.

I don't really care either way, but I don't see why for this type of module we would have a different path pattern than other types of user-specified modules.  Either one is fine but we should just be consistent in the paths.

Reply:
I like the concept of a few "common" locations that can be extended by the user, and agree custom cfe core modules should follow a similar pattern, just seems like the implementation evolved to the point of being due a refactor or at minimum a careful review and documentation.  Rather than pile on I've gone simple for now.  We can fix prior to considering for an actual merge.

Reply:
We can certainly have that discussion about how to clean it up.  There is at least one aspect of that logic that can be simplified.    I don't think that's a reason to bypass it, though.

Problem is - it isn't just that one loop - from there a number of variables/lists are set and other parts of the build system need/expect these patterns to exist, and they won't be correct.  So there is negative fallout from _not_ following the same path/logic.  

Furthermore - the minimal change proposed in #720 to use the same logic is downright trivial compared to all the extra logic proposed here.  I even made it simpler (see commit jphickey/cfe@a921046) - as you don't strictly need the `arch_build.cmake` bits for include_directories - so it literally is only two lines that are actually needed to make this follow the same logic as every other module, which "just works" without the unintended consequences / inconsistencies.

We can talk about ways to clean up the search logic, but in the meantime I would strongly recommend against bypassing it..

Reply:
Note still not transitioned yet.  I'm not following the resistance to using interface libraries, etc but if you've got a commit I can merge that implements your version I'll merge it in before I finish.

Reply:
Updated to built in module framework